paasta_tools/utils.py

```diff
@@ -3013,7 +3013,18 @@ def get_pipeline_config(service: str, soa_dir: str = DEFAULT_SOA_DIR) -> List[Di
 def get_pipeline_deploy_groups(
     service: str, soa_dir: str = DEFAULT_SOA_DIR
 ) -> List[str]:
-    pipeline_steps = [step["step"] for step in get_pipeline_config(service, soa_dir)]
+    pipeline_steps = []
+    for step in get_pipeline_config(service, soa_dir):
+        # added support for parallel steps in a deploy.yaml
+        # parallel steps would break previous functionality as steps arent
+        # expected to be nested in a parallel block
```

Comment:
thoughts on something like
```suggestion
        # steps can show up in one of two ways:
        # 1) inline (e.g., a "top-level" entry like `- step-name-here`)
        # 2) or, in a parallel block - e.g.:
        #    parallel:
        #        - step-name-here
        #        - another-step-name
```

Reply:
oh, and could we add a test for `get_pipeline_deploy_groups()` that uses this new codepath?